File: Lab7/Question9/src/Bird.java
public abstract class Bird extends Animal implements Flyer{
    private int numOfEggs;

    public Bird(String name, int numOfEggs) {
        super(name);
        this.numOfEggs = numOfEggs;
    }

    public int getNumOfEggs() {
        return numOfEggs;
    }

    @Override
    public int getLegs() {
        return 2;
    }
    
    @Override
    public boolean isDangerous() {
    	return false;
    }

    public static void testBird() {
    	System.out.println("Bird is an abstract class, and cannot be initialize.");
    }
}

File: Lab7/Bird.kt
abstract class Bird : Animal, Flyer {
    private val numOfEggs: Int

    constructor(name: String, numOfEggs: Int) : super(name) {
        this.numOfEggs = numOfEggs
    }

    open fun getNumOfEggs(): Int {
        return numOfEggs
    }

    override fun getLegs(): Int {
        return 2
    }

    override fun isDangerous(): Boolean {
        return false
    }
}